### DISCUSSION TITLE
netflow V9 decoding error

### DISCUSSION THREAD

**Discussion by Tayyab146** [akvorado/akvorado#1465] [Q&A]
Hi All,

I'm seeing below not able to find out how to resolve them:

[root@test-vm-10:-/akvorado curl -s http://10.20.5.60:8081/api/ve/inlet/metrics | grep akvorado inlet | grep _error akvorado inlet_core_flows_errors_total(error="SNMP cache miss", exporter="66.xx") 24
akvorado inlet_flow_decoder_errors total(name="netflow") 435
akvorado inlet_flow_decoder_netflow_errors total (error="NetFlow v9 decoding error", exporter="66.xx) 435 akvorado inlet_metadata_provider_snmp_poller_error_requests total(error="get", exporter="66.209.68.242") 15

**Comment by vincentbernat** [comment]
Which ones are increasing?

**Reply by Tayyab146** [reply]
All of them are increasing this is latest:

[root@test-vm-10:~/akvorado# curl -s http://10.20.5.60:8081/api/ve/inlet/metrics | grep akvorado inlet | grep error akvorado inlet_core_flows errors total (error="SNMP cache miss", exporter="66.xx") 159
akvorado inlet_flow_decoder_errors total(name="netflow") 140
akvorado inlet_flow_decoder_netflow_errors total (error="NetFlow v9 decoding error", exporter "66.xx") 140
akvorado inlet metadata_provider_snmp_poller_error_requests total ferror="get", exporter="66.209.68.242") 27

**Reply by Tayyab146** [reply]
@vincentbernat any update what to do?

**Reply by Tayyab146** [reply]
@vincentbernat are you there?

**Reply by vincentbernat** [reply]
I maintain this open-source project in my limited free time. It is not OK to expect a timely response from my side. For your specific needs, I recommend considering professional support services or consultants who can provide dedicated assistance.

**Reply by Tayyab146** [reply]
I'm sorry for that, Can you help with this error when you get a chance, for now only below two errors are increasing:

akvorado inlet_flow_decoder_errors total(name="netflow") 140
akvorado inlet_flow_decoder_netflow_errors total (error="NetFlow v9 decoding error", exporter "66.xx") 140

**Comment by dbray925** [comment]
@vincentbernat thank you again for all your hard work. Looks like we've stumbled onto this same issue:

>akvorado_inlet_flow_decoder_errors_total{name="netflow"} 18
>akvorado_inlet_flow_decoder_netflow_errors_total{error="NetFlow v9 decoding error",exporter="10.10.10.10"} 18

This is coming from a Palo Alto PA-3220 running version 10.2.4-h4. 

Just wanted to subscribe to this topic, and share some additional information we have as well. Please let me know if there is anything I can do to help test or confirm. In the mean time, I'll spin up some other hardware and continue testing.

**Reply by vincentbernat** [reply]
I would need a capture of the netflow packet.

**Reply by dbray925** [reply]
Would the decoded version be good enough, or do you need the full packet capture:

```
Frame 1: 1334 bytes on wire (10672 bits), 1334 bytes captured (10672 bits)
Ethernet II, Src: PaloAltoNetw_00: (##:##:##:##), Dst: ##:##:##:## (##:##:##:##)
Internet Protocol Version 4, Src: 10.10.10.201, Dst: 10.10.10.105
User Datagram Protocol, Src Port: 55175, Dst Port: 9995
Cisco NetFlow/IPFIX
    Version: 9
    Count: 8
    SysUptime: 556712.336000000 seconds
    Timestamp: Nov 14, 2024 10:05:14.000000000 Eastern Standard Time
        CurrentSecs: 1731596714
    FlowSequence: 1248296
    SourceId: 1
   FlowSet 1 [id=257] (1 flows)
        FlowSet Id: (Data) (257)
        FlowSet Length: 159
        [Template Frame: 21]
        Flow 1
            Octets: 60
            Packets: 1
            Protocol: TCP (6)
            IP ToS: 0x00
            TCP Flags: 0x02, SYN
                00.. .... = Reserved: 0x0
                ..0. .... = URG: Not used
                ...0 .... = ACK: Not used
                .... 0... = PSH: Not used
                .... .0.. = RST: Not used
                .... ..1. = SYN: Used
                .... ...0 = FIN: Not used
            SrcPort: 53717 (53717)
            SrcAddr: #.#.#.#
            InputInt: 5
            DstPort: 2638 (2638)
            DstAddr: #.#.#.#
            OutputInt: 0
            [Duration: 0.000000000 seconds (switched)]
                StartTime: 560628.336000000 seconds
                EndTime: 560628.336000000 seconds
            ICMP Type: 0x0000
            Direction: Ingress (0)
            Flow Id: 0
            Firewall Event: Flow denied (3)
            Private Enterprise Number: 25461
            Unknown Field Type: Type 56701: Value (hex bytes): 6e 6f 74 2d 61 70 70 6c 69 63 61 62 6c 65 00 00 00 00 00 00 00 00 00 00 …
            Unknown Field Type: Type 56702: Value (hex bytes): 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 …

```

**Comment by dbray925** [comment]
My initial response had the "No Template Found" issue, which is discussed here: https://knowledgebase.paloaltonetworks.com/KCSArticleDetail?id=kA14u0000008UuYCAU I edited the above with a better decoded cflow. Please let me know if that is not good enough, and/or if you need the full capture file. Thank you.

**Reply by vincentbernat** [reply]
Check the logs instead to know the exact decoding error. You should have a "error while decoding NetFlow v9" with a more precise message.
